src/lib/certificate-manager.ts

```diff
@@ -148,11 +152,67 @@ export async function openCertificate() {
 	shell.showItemInFolder( CA_CERT_PATH );
 }
 
+/**
+ * Checks if the root CA certificate is already trusted by the system
+ * @returns A promise that resolves to true if the certificate is trusted, false otherwise
+ */
+export async function isRootCATrusted(): Promise< boolean > {
+	if ( ! fs.existsSync( CA_CERT_PATH ) ) {
+		return false;
+	}
+
+	if ( process.platform === 'win32' ) {
+		try {
+			// Execute certutil with more specific validation
+			const { stdout } = await execFilePromise( 'certutil', [
+				'-verify',
+				'-urlfetch',
+				'-purpose',
```

Comment:
I am getting the unknown option `-purpose` error on Windows 11 Pro. It needs some refactoring.